core/src/main/java/org/infinispan/commands/CommandsFactoryImpl.java

```diff
@@ -544,8 +547,43 @@ public void initializeReplicableCommand(ReplicableCommand c, boolean isRemote) {
             InvalidateVersionsCommand invalidateVersionsCommand = (InvalidateVersionsCommand) c;
             invalidateVersionsCommand.init(dataContainer, orderedUpdatesManager, stateTransferLock, stateTransferManager);
             break;
+         // === Functional commands ====
+         case ReadOnlyKeyCommand.COMMAND_ID:
+            ((ReadOnlyKeyCommand)c).init(componentRegistry);
```

Comment:
Could you squash these cases (casting to `AbstractWriteKeyCommand` or `AbstractWriteManyCommand`)?

Reply:
All the method is like that, you mean on the default part see if it's instanceOf AbstractWriteKeyCommand or the other to call the init method ?